zio-http/src/main/scala/zhttp/http/Http.scala

```diff
@@ -843,6 +862,11 @@ object Http {
 
   private case class Attempt[A](a: () => A) extends Http[Any, Nothing, Any, A]
 
+  private final case class Combine[R, E, EE, A, B, BB](
```

Comment:
A better signature for combine could be
```scala
final case class Combine[R, R1 <: R, E, EE >: E, A, B, BB >: B](
  self: Http[R, E, A, B],
  other: Http[R1, EE, A, BB],
) extends Http[R1, EE, A, BB]
```

this could be better represented as:
```scala
val a = Http.collectZIO[Int] {
  case 1 => ZIO.fail(new InterruptedException())
  case 2 => ZIO.environment[Int]
}

val b = Http.collectZIO[Int] {
  case 1 => ZIO.fail(new RuntimeException())
  case 4 => ZIO.environment[String]
}
val r: Combine[Int, String with Int, InterruptedException, Exception, Int, Int, Any] = Combine(a, b)
```